src/pages/tutorial/cw-contract/query.mdx

```diff
@@ -0,0 +1,323 @@
+# Creating a query
+
+We have already created a simple contract reacting to an empty instantiate message. Unfortunately,
+it is not very useful. Let's make it a bit reactive.
+
+First, we need to add [`serde`](https://crates.io/crates/serde) crate to our dependencies. It would
+help us with the serialization and deserialization of query messages.
+
+```toml {11} copy filename="Cargo.toml"
+[package]
+name = "contract"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+cosmwasm-std = { version = "2.1.4", features = ["staking"] }
+serde = { version = "1.0.214", default-features = false, features = ["derive"] }
+```
+
+Now go add a new query entry point:
+
+```rust {7-10, 22-29} copy filename="src/lib.rs" template="empty"
+use cosmwasm_std::{
+    entry_point, to_json_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo,
+    Response, StdResult,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
```

Comment:
It's kind of better to use this macro for raw contracts:
https://docs.rs/cosmwasm-schema/latest/cosmwasm_schema/attr.cw_serde.html

Reply:
I agree, but as I understand, the intention of Bart when he was writing this book was to in minimal steps introduce new concepts.

The `cw_serde` is introduced in the https://book.cosmwasm.com/basics/good-practices.html#json-schema page.

You can consider this and if you still think it's better to use `cw_serde` right away I will change it.

Reply:
Alright, I guess it's not likely to cause much damage.